File: GameEngine_IOS/GameEngine/Commons/Vector3f.m
#import "Vector3f.h"

@implementation Vector3f


- (id)init : (float) aX : (float) aY : (float) aZ
{
    self = [super init : aX : aY];
    if (self) {
        [self setZ: aZ];
    }
    return self;
}

@end

File: GameEngine_Android/commons/src/main/java/com/dferreira/commons/Vector3f.java
package com.dferreira.commons;

/**
 * Represents a 3D Vector with their x,y,z coordinates
 */
public class Vector3f extends Vector2f {

    /* The coordinates them selves */
    public float z;

    /**
     * The constructor of the vector
     *
     * @param x
     *            x-coordinate
     * @param y
     *            y-coordinate
     * @param z
     *            z-coordinate
     */
    public Vector3f(float x, float y, float z) {
        super(x, y);
        this.z = z;
    }

    /**
     * @return A string describing the vector3f
     */
    @Override
    public String toString() {
        return "Vector3f{" +
                "x=" + x +
                "y=" + y +
                "z=" + z +
                '}';
    }
}
